docs/rules/require-expect.md

```diff
@@ -0,0 +1,84 @@
+# Ensure that `expect` is called when using assertions outside of top-level test callback
+
+QUnit's `assert.expect(...)` helps developers create tests that correctly fail
+when their expected number of assertions are not called. QUnit will throw an
+error if no assertions are called by the time the test ends. This rule goes
+further and ensures that `expect` is used anytime a developer uses are assertion
+that is nested in a block or anytime a developer passes `assert` to another
+function.
```

Comment:
Same as above-- we can simplify this to just say we check for expect at linting time. Although this would be the first place we could mention the option of only enforcing it when the test does not have full control over assertions. Let me know if you want suggestions from me on wording or whatnot.